addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -388,23 +384,25 @@ public void pressButton(String device, String button) {
         }
     }
 
-    /**
-     * Cache a call to get the harmony config so we don't overwhelm it with requests.
-     *
-     * @return
-     */
-    public synchronized HarmonyConfig getCachedConfig() {
-        if (client == null) {
-            return null;
-        }
-
-        Date now = new Date();
-        if (cachedConfig == null || cacheConfigExpireDate == null || now.after(cacheConfigExpireDate)) {
-            logger.debug("Refreshing conf cache");
-            cachedConfig = client.getConfig();
-            cacheConfigExpireDate = new Date(System.currentTimeMillis() + CONFIG_CACHE_TIME);
-        }
-        return cachedConfig;
+    public synchronized CompletableFuture<HarmonyConfig> getConfigFuture() {
+        return configCache.getValue(() -> {
+            return new CompletableFuture<HarmonyConfig>() {
+                {
+                    if (client == null) {
+                        logger.debug("Could not get config because client is null");
+                        complete(null);
+                    } else {
+                        try {
+                            logger.debug("Getting config from client");
+                            complete(client.getConfig());
+                        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
It looks like most of the time only RuntimeExceptions are thrown by the client. But I'd rather only change this after more thorough testing.

Reply:
Agree lets go for this solution now.